src/TgSharp.Core/TelegramClient.cs

```diff
@@ -57,14 +58,16 @@ public Session Session
             TcpClientConnectionHandler handler = null
             )
         {
-            if (apiId == default(int))
+            if (apiId == default)
                 throw new MissingApiConfigurationException("API_ID");
+
             if (string.IsNullOrEmpty(apiHash))
                 throw new MissingApiConfigurationException("API_HASH");
 
-            if (store == null)
-                store = new FileSessionStore();
-            this.store = store;
+            this.store = store ?? new JsonFileSessionStore();
+
+            if (store == null && File.Exists($"{sessionUserId}.dat"))
```

Comment:
I think this is a bit confusing because of the name clash between the parameter `store` and the field `this.store`, it would be more clear if we do this:

```
if (store != null) {
    this.store = store;
} else {
    this.store = new JsonFileSessionStore();
    if (File.Exists($"{sessionUserId}.dat")) {
        MigrateOldSessionFile(sessionUserId);
    }
}
```

Reply:
Agreed, it is confusing.

Reply:
Switched to your code tho, way cleaner and less confusing. Let me know if there's anything else you'd like to discuss about, then I'll push.

Reply:
ah right, if you do it inverted, you can even save one line:

```
if (store == null) {
    store = new JsonFileSessionStore();
    if (File.Exists($"{sessionUserId}.dat")) {
        store = MigrateOldSessionFile(sessionUserId);
    }
}
this.store = store;
```

Reply:
You can't, `MigrateOldSessionFile` uses `this.store` so it must be set before we call the method. Nice try tho 😄

Reply:
Unless you want to do that inside of `MigrateOldSessionFile`, and return an `ISessionStore` from it?

Reply:
```cs
if (store == null && File.Exists($"{sessionUserId}.dat"))
{
    store = MigrateOldSessionFile(sessionUserId);
}

this.store = store;
```
```cs
private ISessionStore MigrateOldSessionFile(string sessionUserId)
{
    var jsonSessionStore = new JsonFileSessionStore();
    var fileSessionStore = new FileSessionStore();

    var session = fileSessionStore.Load(sessionUserId);

    jsonSessionStore.Save(session);

    File.Delete($"{sessionUserId}.dat");

    return jsonSessionStore;
}

```

Reply:
Like that:
```cs
if (store == null)
{
    store = File.Exists($"{sessionUserId}.dat") ? MigrateOldSessionFile(sessionUserId) : new JsonFileSessionStore();
}

this.store = store;
```
```cs
private ISessionStore MigrateOldSessionFile(string sessionUserId)
{
    var jsonSessionStore = new JsonFileSessionStore();
    var fileSessionStore = new FileSessionStore();

    var session = fileSessionStore.Load(sessionUserId);

    jsonSessionStore.Save(session);

    File.Delete($"{sessionUserId}.dat");

    return jsonSessionStore;
}
```
Doesn't seem better tho.

Reply:
I like it more, and you can make MigrateOldSessionFile static

Reply:
Let's use this:
```cs
if (store == null)
{
    this.store = new JsonFileSessionStore();

    if (File.Exists($"{sessionUserId}.dat"))
    {
        MigrateOldSessionFile(sessionUserId);
    }
}
else
{
    this.store = store;
}
```
```cs
private void MigrateOldSessionFile(string sessionUserId)
{
    var fileSessionStore = new FileSessionStore();

    var session = fileSessionStore.Load(sessionUserId);

    store.Save(session);

    File.Delete($"{sessionUserId}.dat");
}
```

Reply:
this also makes me think that FilleSessionStore.Load should be static

Reply:
It can't be static? `Load` is a method from the `ISessionStore` interface